### ISSUE TITLE
The FileStore closer always return error

### ISSUE THREAD

**Issue by jotadrilo** [oras-project/oras-go#84]
## Description

I have found that `FileStore` implements a buggy `Close` function:

https://github.com/oras-project/oras-go/blob/99e2fdec2e997165c48148739767530965aeac26/pkg/content/file.go#L306-L316

### Example

https://go.dev/play/p/UvfCFZTUwpk

```golang
package main

import (
	"fmt"
	"log"
	"strings"

	"github.com/pkg/errors"
)

func main() {
	errs := []string{}
	err := errors.New(strings.Join(errs, "; "))
	if err != nil {
		fmt.Printf("This is returning an error due to %v\n", errs)
		log.Fatal(err)
	}
}
```

Run

```
This is returning an error due to []
2009/11/10 23:00:00 
```

## Affected users

Only users handling errors in the deferred function:

```golang
# not affected
defer store.Close()

# affected
defer func() {
	err := store.Close()
	# e is the named returned variable of the function
	if e == nil {
		e == err
	}
}()
```

### Workaround

```golang
defer func() {
	err := store.Close()
	# e is the named returned variable of the function
	if e == nil && err != nil && err.Error() != "" {
		e = err
	}
}()
```

## Fix proposal

Check errors slice before returning a new (valid) error:

```golang
// Close frees up resources used by the file store 
func (s *File) Close() error { 
	var errs []string 
	s.tmpFiles.Range(func(name, _ interface{}) bool { 
		if err := os.Remove(name.(string)); err != nil { 
			errs = append(errs, err.Error()) 
		} 
		return true 
	})
	if len(errs) == 0 {
		return nil
	}
	return errors.New(strings.Join(errs, "; ")) 
} 
```

**Comment by shizhMSFT** [comment]
@jdolitsky @Wwwsylvia This is related to `v1`.

**Comment by shizhMSFT** [comment]
Thanks @jotadrilo for pointing out the bug. I've converted your proposal into a PR #93. Since the `/pkg` folder will be eventually removed in `main`, I will hold the change to the `main` branch.

**Comment by shizhMSFT** [comment]
Closing as the fix #93 has been merged to the `v1` branch.
